### PR TITLE
Add tests for English auction

### PR DESCRIPTION
**Pull request by vgantchev**
WIP, more info to follow

### CODE REVIEW

File: pallets/auctions/src/lib.rs [hunk 1]
```diff
@@ -194,13 +220,26 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
 		fn on_finalize(now: T::BlockNumber) {
-			Self::conclude_auction(now);
+			Self::conclude_auction(now).unwrap_or_default();
```
Comment (green-jay) [inline pallets/auctions/src/lib.rs:223]: what are you trying to do? if you want to pass the result of a function to the outer scope use the question mark. unwrap_or_default should be used only when you actually want to pass the default if unwrap fails

**Review by green-jay — CHANGES_REQUESTED** [review]
remove unwrap and merge

File: pallets/auctions/src/lib.rs (hunk 1 above)
Comment (enthusiastmartin) [reply pallets/auctions/src/lib.rs:223]: Note: on finalize cannot fail. And you cannot return any error from there.

Comment (enthusiastmartin) [reply pallets/auctions/src/lib.rs:223]: And you cannot even return weight consumed by on finalized. That's why you must return return the wieght of on finalize ffrom on_initialized. 

Which i cannot see in this pallet ?! 

You need to benchmarks somehow the weight used in on finalized and return it from on_initialized.

**Review by enthusiastmartin — CHANGES_REQUESTED** [review]
Auction trait:  you can remove associated type AccountId from the trait. You provide it as generic too, so it is useless having it as associated type as well as it does nothing.

You need to benchmark on finalize and return weight from on initialized.

File: pallets/auctions/src/lib.rs [hunk 2]
```diff
@@ -183,9 +199,19 @@ pub mod pallet {
 
 		#[pallet::weight(<T as Config>::WeightInfo::delete_auction())]
 		pub fn delete_auction(origin: OriginFor<T>, id: T::AuctionId) -> DispatchResultWithPostInfo {
```
Comment (enthusiastmartin) [inline pallets/auctions/src/lib.rs:201]: It is laready mixed use of DispatchResult and DispatchResultWithPostInfo in this pallet.


Following refactor #111 , it is actaully better to use DispatchResult unless you need specifically the post info.

Applies to each dispatchable in this pallet.

```suggestion
		pub fn delete_auction(origin: OriginFor<T>, id: T::AuctionId) -> DispatchResult {
```

File: pallets/auctions/src/lib.rs [hunk 3]
```diff
@@ -194,13 +220,26 @@ pub mod pallet {
...
 		fn on_finalize(now: T::BlockNumber) {
-			Self::conclude_auction(now);
+			Self::conclude_auction(now).unwrap_or_default();
 		}
 	}
 }
 
 impl<T: Config> Pallet<T> {
-	fn conclude_auction(now: T::BlockNumber) {
+	fn validate_update_delete(
```
Comment (enthusiastmartin) [inline pallets/auctions/src/lib.rs:229]: The name is confusing to me . ? What it does - validates, updates or deletes?:-)

According to implementation - i guess it is something like "can_delete" ?  If so, it does not need DispatchResult, simple bool type would suffice.

**Review by enthusiastmartin — COMMENTED** [review]
Few additional findings.

**Comment by vgantchev** [comment]
@enthusiastmartin thx for review. there is another wip branch with the refactor. lets focus this pr on tests so i can rebase and open the refactor pr

**Comment by vgantchev** [comment]
closing in favour of https://github.com/galacticcouncil/Basilisk-node/pull/233

**Comment by vgantchev** [comment]
closing in favour of #233

**Pull request closed without merging**